Snowflake.Data/Core/Session/SFSession.cs

```diff
@@ -113,7 +125,17 @@ internal void ProcessLoginResponse(LoginResponse authnResponse)
                     "");
 
                 logger.Error("Authentication failed", e);
-                throw e;
+
+                if (e.ErrorCode == SFError.ID_TOKEN_INVALID.GetAttribute<SFErrorAttr>().errorCode)
+                {
+                    logger.Error("SSO Token has expired or not valid. Reauthenticating without SSO token...", e);

```

Comment:
I think that Info would be more appropriate here as token expiration is expected scenario and does not generate any issue within the driver.

Reply:
That makes sense, it's now changed to Info